distributed/executor.py

```diff
@@ -193,6 +193,22 @@ def release(self):
             self._cleared = True
             self.executor._dec_ref(tokey(self.key))
 
+    def __getstate__(self):
+        state = self.__dict__.copy()
+        state.update({'event': self.event.is_set(), 'status': self.status})
+        del state['executor']
+        return state
+
+    def __setstate__(self, state):
+        self.__dict__ = state
+        e = default_executor()
+        event = Event()
+        if state.pop('event'):
+            event.set()
+        e.futures[tokey(state['key'])] = {'event': event,
+                                          'status': state.pop('status')}
+        state['executor'] = e
```

Comment:
OK, so the issue was that the call to update-graph from the second executor was happening before a yield had ensured that the call from the first executor was actually processed in the scheduler (sot hey key didn't exist inthe state dicts).
I'm not sure that this is a case we need to worry about, as the second executor cannot have enough information to generate the future by itself, it should only be called on things that the scheduler knows about.